### PR TITLE
Match muted behavior with JS SDK

### PR DESCRIPTION
**Pull request by hiroshihorie**

### CODE REVIEW

File: lib/src/track/local_track_publication.dart [hunk 1]
```diff
@@ -1,6 +1,5 @@
-import '../events.dart';
-import '../extensions.dart';
-import '../logger.dart';
+import 'package:livekit_client/src/track/local_track.dart';
```
Comment (davidzhao) [inline lib/src/track/local_track_publication.dart:1]:
```suggestion
import 'local_track.dart';
```

File: lib/src/track/local_track_publication.dart [hunk 2]
```diff
@@ -22,28 +21,39 @@ class LocalTrackPublication extends TrackPublication {
     });
   }
 
-  /// Mute or unmute the current track. When muted, track will stop sending data
-  @override
-  set muted(bool val) {
-    if (val == muted) return;
-    logger.finer('setMute: ${val}');
+  // /// Mute or unmute the current track. When muted, track will stop sending data
```
Comment (davidzhao) [inline lib/src/track/local_track_publication.dart:24]: delete the commented out code?

File: lib/src/track/local_track.dart [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import 'package:livekit_client/livekit_client.dart';
```
Comment (davidzhao) [inline lib/src/track/local_track.dart:1]: this should be relative import as well correct?

File: lib/src/track/local_track_publication.dart [hunk 4]
```diff
@@ -22,28 +21,39 @@ class LocalTrackPublication extends TrackPublication {
...
+  //   if (val == muted) return;
+  //   logger.finer('setMute: ${val}');
+
+  //   super.muted = val;
+  //   track?.mediaStreamTrack.enabled = !val;
+  //   _participant.engine.signalClient.sendMuteTrack(sid, val);
 
-    super.muted = val;
-    track?.mediaStreamTrack.enabled = !val;
-    _participant.engine.signalClient.sendMuteTrack(sid, val);
```
Comment (davidzhao) [inline lib/src/track/local_track_publication.dart:33]: what is updating the server to tell it the track is muted?

**Comment by davidzhao** [comment]
lol, just saw the WIP.. oops. I probably jumped the gun when reviewing. converted this to a draft PR

**Comment by hiroshihorie** [comment]
Oh this is still WIP haha will continue later today 😅

**Comment by hiroshihorie** [comment]
I will test it 🙂

File: lib/src/track/local_track_publication.dart [hunk 5]
```diff
@@ -1,8 +1,9 @@
 import '../events.dart';
 import '../extensions.dart';
-import '../logger.dart';
+import '../internal/events.dart';
 import '../participant/local_participant.dart';
 import '../proto/livekit_models.pb.dart' as lk_models;
+import '../track/local_track.dart';
```
Comment (davidzhao) [inline lib/src/track/local_track_publication.dart:6]:
```suggestion
import 'local_track.dart';
```

**Review by davidzhao — APPROVED** [review]
lgtm!

**Pull request merged**